### PR TITLE
More buildings mod items & monsters

### LINKED ISSUE #22225: "More locations" mod adds a building with unsupported tiles
Running `cataclysm --check-mods` results in:
```
src/overmap.cpp:4814 [void building_bin::finalize()] Tried to add city building small_storage_units, but it has a part with non-zero x or y coords (not supported yet)
```

The building is then silently dropped, so it has no in-game effects.

### PR DESCRIPTION
**Pull request by laeaton2001**
Fixes #22225 <strike>and should help with #20159.</strike>
Changes:
- Updated small storage units id to stop conflict with city store.
- Moved baseball field & shopping plaza to use palletes
- added loot and monster spawns to:
  - baseball field
  - shopping plaza
  - daycare
  - reststop

Not 100% sure on balance so comments welcome.

### CODE REVIEW

File: data/mods/More_Buildings/json/daycare.json [hunk 1]
```diff
@@ -106,6 +106,11 @@
       "place_items": [
         { "item": "daycare_misc", "x": [ 3, 7 ], "y": [ 12, 16 ], "chance": 35, "repeat": [ 1, 5 ] },
         { "item": "daycare_kitchen", "x": 21, "y": [ 20, 21 ], "chance": 80, "repeat": [ 1, 5 ] }
+      ],
+      "place_monsters": [
+        { "monster": "GROUP_MAYBE_ZOMBIE", "x": [ 13, 20 ], "y": 19 },
+		{ "monster": "GROUP_SCHOOL", "x": [ 3, 7 ], "y": [ 12, 16 ] },
+		{ "monster": "GROUP_SCHOOL", "x": [ 3, 10 ], "y": [ 4, 8 ] }
```
Comment (Coolthulhu) [inline data/mods/More_Buildings/json/daycare.json]: Tabs sneaked in.

File: data/mods/More_Buildings/json/map_palletes.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
+[
+  {
+    "type": "palette",
+    "id": "baseballfield_pal",
```
Comment (Coolthulhu) [inline data/mods/More_Buildings/json/map_palletes.json]: That "_pal" is unnecessary, it's obvious from context that it is a palette.

**Comment by ZhilkinSerg** [comment]
Actually that won't help with #20159 - `More buildings` and `More locations` are two different mods.

**Comment by laeaton2001** [comment]
Whoops misread the mod in 20159, you're right this won't help that at all.

**Comment by laeaton2001** [comment]
Can anyone shed some light on what is failing in the build?
make: *** [data/mods/More_Buildings/json/restparking_1.json] Error 1
make: INTERNAL: Exiting with 4 jobserver tokens available; should be 3!

**Comment by kevingranade** [comment]
The line above what you pasted makes it clear
```
Formatted data/mods/More_Buildings/json/restparking_1.json
```
Run ```make style-json``` locally and commit the result.

**Pull request merged**